app/web_framework/cluster_master_application.py

```diff
@@ -152,6 +156,51 @@ def get(self, build_id, subjob_id, atom_id):
         self.write(response)
 
 
+class _AtomConsoleHandler(_ClusterMasterBaseAPIHandler):
```

Comment:
Ah, yeah I agree with you about placing this logic here -- it's all pretty webby. I don't really see much that could be moved into `ClusterMaster`. It makes sense though -- this is probably some of the heaviest web-centric logic that we've added so far.